pysiaf/aperture.py

```
@@ -409,6 +409,27 @@ def get_polynomial_coefficients(self):
                     [getattr(self, s) for s in DISTORTION_ATTRIBUTES if seed in s])[
                              0:number_of_coefficients]
             return dict
+    def set_polynomial_coefficients(self, sci2idlx, sci2idly, idl2scix, idl2sciy):
+        """Place a full set of coefficients into aperture
+        Input
+        : param sci2idlx - array of Sci2IdlX coefficients in JWST order
+        : param sci2idly - array of Sci2IdlY coefficients
+        : param idl2scix - array of Idl2SciX coefficients
+        : param idl2sciy - array of Idl2SciY coefficients
+        """
+
+        if self.Sci2IdlDeg is None:
+            return None
```

Instead of silently returning, please raise an error.